clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -573,8 +572,13 @@ public void onSuccess(ClientResponse clientResponse, RequestFuture<T> future) {
                 return;
             }
 
-            R response = parse(clientResponse);
-            handle(response, future);
+            try {
+                R response = parse(clientResponse);
+                handle(response, future);
+            } catch (RuntimeException e) {
```

Comment:
It could be a SchemaException or an unexpected exception thrown from the handler. It seemed safest to catch RuntimeException to ensure that the future was always completed. Otherwise, you could end up stuck in poll(future) forever. I was debating whether we should also add a finally block to ensure we didn't miss any cases in the handlers.